ocaml/sdk-gen/component-test/jsonrpc-client/go/vm_test.go

```diff
@@ -0,0 +1,494 @@
+		t.Fail()
+		return
+	}
+}
+
+func TestVMGetSnapshotTime(t *testing.T) {
+	data, err := ReadJsonFile("../../spec/xapi-24/vm_get_snapshot_time.json")
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+
+	type ResultBody struct {
+		Result time.Time `json:"result,omitempty"`
+	}
+
+	type MethodResult struct {
+		MethodName ResultBody `json:"VM.get_snapshot_time,omitempty"`
+	}
+
+	type TestResponse struct {
+		Method         []string               `json:"method,omitempty"`
+		Params         map[string]interface{} `json:"params,omitempty"`
+		ExpectedResult MethodResult           `json:"expected_result"`
+	}
+
+	type TestResult struct {
+		Key TestResponse `json:"xapi-24/vm_get_snapshot_time_07"`
+	}
+
+	var expectData TestResult
+	if err := json.Unmarshal(data, &expectData); err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+
+	expectedResult := expectData.Key.ExpectedResult.MethodName.Result
+
+	session, err := GetSession("xapi-24/vm_get_snapshot_time_07")
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+	var vmRef xenapi.VMRef = "OpaqueRef:6ef08bce-0bf0-30ff-804f-5f0ee4bbdd13"
+	result, err := xenapi.VM.GetSnapshotTime(session, vmRef)
+
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+
+	if result != expectedResult {
+		t.Log("The result returned not the same with expected -> The XAPI outcome diverges from the anticipated value.")
+		t.Fail()
+		return
+	}
+}
+
+func TestVMQueryDataSource(t *testing.T) {
+	session, err := GetSession("xapi-24/vm_query_data_source_08")
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+
+	var vmRef xenapi.VMRef = "OpaqueRef:6ef08bce-0bf0-30ff-804f-5f0ee4bbdd13"
+
+	result, err := xenapi.VM.QueryDataSource(session, vmRef, "CPU0 usage")
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+	if !math.IsNaN(result) {
+		t.Log("The result returned not the same with expected -> The XAPI outcome diverges from the anticipated value.")
+		t.Fail()
+		return
+	}
+}
+
+func TestVMAsyncImport(t *testing.T) {
+	data, err := ReadJsonFile("../../spec/xapi-24/async_vm_import.json")
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+
+	type ResultBody struct {
+		Result xenapi.TaskRef `json:"result,omitempty"`
+	}
+
+	type MethodResult struct {
+		MethodName ResultBody `json:"Async.VM.import"`
+	}
+
+	type TestResponse struct {
+		Method         []string               `json:"method,omitempty"`
+		Params         map[string]interface{} `json:"params,omitempty"`
+		ExpectedResult MethodResult           `json:"expected_result"`
+	}
+
+	type TestResult struct {
+		Key TestResponse `json:"xapi-24/async_vm_import_12"`
+	}
+
+	var expectData TestResult
+	if err := json.Unmarshal(data, &expectData); err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+
+	expectedResult := expectData.Key.ExpectedResult.MethodName.Result
+
+	session, err := GetSession("xapi-24/async_vm_import_12")
```

Comment:
Ideally, the test ID can be get from the `data` rather than hard code in two places here.

Reply:
Updated